### DISCUSSION TITLE
ATTACH performance in 1.1.0

### DISCUSSION THREAD

**Discussion by kirankpandi** [duckdb/duckdb#14702] [General]
Has there been any change in the implementation of the `ATTACH` statement from version 1.0.0 to 1.1.0? We’ve observed that in version 1.1.0, attaching files—particularly those with a large number of tables—takes more than twice as long compared to version 1.0.0. Curious if there were any significant updates or modifications affecting this performance?

**Comment by carlopi** [comment]
Do you have a reproducible test-case that you can share? I am not able to respond, but if you have something where measuring this it could be handy to dig down and discover whether it can be improved.

Also note we are now at v1.1.3, situation might have already have changed (either way :) )

**Comment by kirankpandi** [comment]
Even with `1.1.3` I see similar behavior as `1.1.0`. Here is the db file I used - https://drive.google.com/file/d/1l_3bK7VmLWRRMCp51nmuLh1QQvPJxNp_/view?usp=sharing. And below is the code I used. Note that time it takes to attach the file is less than 100ms. But you can clearly see an increase from `1.0.0` to `1.1.0`.

```
public class TestAttach {

    public void testAttachTiming() throws SQLException {
        var connecProps = new Properties();
        try(var conn = DriverManager.getConnection("jdbc:duckdb:", connecProps);
            var stmt = conn.createStatement()) {
            printVersion(stmt);
            var startTime = System.currentTimeMillis();
            stmt.execute("ATTACH '/work/data/attach_perf_data.db' AS file_db");
            var timeElapsed = System.currentTimeMillis() - startTime;
            System.out.println("Time elapsed = " + timeElapsed);
        }
    }

    private void printVersion(Statement stmt) throws SQLException {
        try(var rs = stmt.executeQuery("select version()")) {
            rs.next();
            System.out.println("DuckDB version = " + rs.getString(1));
        }
    }


    public static void main(String[] args) throws SQLException {
        new TestAttach().testAttachTiming();
    }
}

```
